landing_page: Add label for next and previous arrow buttons.

TanishaShrotriya96 opened the pull request:
[#17467](https://github.com/zulip/zulip/issues/17467)

**Testing plan:**  Manually tested as this is a small feature update, haven't written Puppeteer tests, need guidance here if tests are required

**GIFs or screenshots:** 
Labels appear upon hover.

![next_label](https://user-images.githubusercontent.com/36158530/112732655-3e9cdf80-8f61-11eb-8969-566a928924a9.png)

![prev_label](https://user-images.githubusercontent.com/36158530/112732657-4066a300-8f61-11eb-9273-e6c122bcf16a.png)

Riken-Shah commented on static/styles/portico/landing_page.css:
```diff
@@ -3863,3 +3863,17 @@ nav {
     padding-left: 10px;
 }
+
+.carousel-control.left:hover #prev {
```
Your CSS code shouldn't be here it should inside the `.tour .carousel-control` block (on line `1553`).

Riken-Shah reviewed (commented):
@TanishaShrotriya96 Posted a comment.
We don't run puppeteer tests on the landing page.
One small suggestion, You can improve your commit message to describe what you solving more clearly. 
Something like,  `landing-page: Show next and previous labels when hover on arrow buttons.`

Riken-Shah commented on static/styles/portico/landing_page.css:
```diff
@@ -1573,6 +1573,20 @@ nav {
     }
 }
 
+.carousel-control.left:hover #prev {
```
I meant inside the block.
Something like this,
```css
.tour .carousel-control { /* Line 1533 */
    background: 0;
    border: 0;
    color: hsl(223, 13%, 52%);
    font-size: 2em;
    top: 50%;
    font-weight: 400;
    opacity: 1;
    border-radius: 0;

    &:hover {
        color: hsl(220, 23%, 33%);
    }
+
+    .carousel-label {
+        display: none;
+        font-family: inherit;
+        font-size: 14px;
+    }

```

TanishaShrotriya96 replied:
oh, I see...

TanishaShrotriya96 replied:
What about the css for hover, that should remain outside right?

Riken-Shah replied:
No see next [comment](https://github.com/zulip/zulip/pull/17844#discussion_r602848985).

Riken-Shah commented on static/styles/portico/landing_page.css:
```diff
@@ -1573,6 +1573,20 @@ nav {
+    display: block;
+}
+
+.carousel-control.right:hover #next {
```
Also you don't need `.carousel-control.left:hover #prev` and `.carousel-control.right:hover #next`,
you can just instead,
```css
  &:hover { /* Line 1563 */
+       display: block;
        color: hsl(220, 23%, 33%);
    }
```

TanishaShrotriya96 replied:
Is this what you meant..? It isn't working for me locally. 

``` css
.tour .carousel-control {  /* Line 1553 */
    background: 0;
    border: 0;
    color: hsl(223, 13%, 52%);
    font-size: 2em;
    top: 50%;
    font-weight: 400;
    opacity: 1;
    border-radius: 0;
    
+  .carousel-label {
+       display: none;
+       font-family: inherit;
+       font-size: 14px;
+   }
     
    &.right {
        right: 40px;
    }
    
    &.left {
        left: 40px;
    }
    
    &:hover {
+       display: block;
        color: hsl(220, 23%, 33%);

    }

}
```

Riken-Shah replied:
Ohh! My bad try this,
```css
.tour .carousel-control {  /* Line 1553 */
    background: 0;
    border: 0;
    color: hsl(223, 13%, 52%);
    font-size: 2em;
    top: 50%;
    font-weight: 400;
    opacity: 1;
    border-radius: 0;
    
+  .carousel-label {
+       display: none;
+       font-family: inherit;
+       font-size: 14px;
+   }
     
    &.right {
        right: 40px;
    }
    
    &.left {
        left: 40px;
    }
    
    &:hover {
        color: hsl(220, 23%, 33%);
+        .carousel-label {
+        display: block;
+        }
    }

}
```

TanishaShrotriya96 replied:
Yep works! Thanks

Riken-Shah reviewed (commented):
@TanishaShrotriya96 Posted some more comments.

TanishaShrotriya96 commented:
> @TanishaShrotriya96 Posted a comment.
> We don't run puppeteer tests on the landing page.
> One small suggestion, You can improve your commit message to describe what you solving more clearly.
> Something like, `landing-page: Show next and previous labels when hover on arrow buttons.`

Thanks for letting me know!

Riken-Shah commented on templates/zerver/hello.html:
```diff
@@ -42,7 +42,6 @@ <h1>Chat for distributed teams.</h1>
                         <div class="item-container">
                             <div class="item active">
                                 <div class="item-inner">
-
```
Why did you delete the empty line here?

TanishaShrotriya96 replied:
I didn't edit anything near line 45, but I did run prettier, so that could have been why an empty line was removed. I hope no code was lost?

TanishaShrotriya96 replied:
I ran this command -

`./tools/lint --fix
`
As it had failed in a test earlier.

Riken-Shah replied:
Yeah, I guess then it's alright.

TanishaShrotriya96 replied:
> Now, Everything LGTM.

Yay!!

Riken-Shah commented:
Now, Everything LGTM.

timabbott commented:
Hmm, I think we should probably do this with the standard `aria-label` and `title` elements, rather than doing something custom that won't be visible to screenreaders for accessibility.

TanishaShrotriya96 commented:
> Hmm, I think we should probably do this with the standard `aria-label` and `title` elements, rather than doing something custom that won't be visible to screenreaders for accessibility.

@timabbott , I tried the following code as per your suggestion -

``` HTML
<!-- line 143 in hello.html-->
<a class="carousel-control left visibility-control hide" href="#tour-carousel" data-slide="prev" aria-labelledby="previous" title="Previous"><i class="fa fa-chevron-left" aria-hidden="true"></i></a>
```
I am getting an error with prettier -
```bash
html      | `title` value should be translatable. at templates/zerver/hello.html line 143:
html      | <a class="carousel-control left visibility-control hide" href="#tour-carousel" data-slide="prev" aria-labelledby="previous" title="Previous"><i class="fa fa-chevron-left" aria-hidden="true"></i></a>
```
I tried a few things, but the error persisted.

1. Added a `translate` attribute.
2. Moved the title attribute to the `<i>` tag.

Also, I couldn't find another instance in the zulip codebase where a tag was using ` aria-labelledby ` as well as a `title` attribute.
Is there something I may have missed?

Riken-Shah commented:
Try this,
```html
<a class="carousel-control left visibility-control hide" href="#tour-carousel" data-slide="prev" aria-labelledby="previous" title="{{ _('Previous') }}"><i class="fa fa-chevron-left" aria-hidden="true"></i></a>
```
You need to use `_()` function so that string can translate,
Read more about it [here](https://zulip.readthedocs.io/en/latest/translating/internationalization.html#backend-translations).

TanishaShrotriya96 commented:
> Try this,
> 
> ```
> <a class="carousel-control left visibility-control hide" href="#tour-carousel" data-slide="prev" aria-labelledby="previous" title="{{ _('Previous') }}"><i class="fa fa-chevron-left" aria-hidden="true"></i></a>
> ```
> 
> You need to use `_()` function so that string can translate,
> Read more about it [here](https://zulip.readthedocs.io/en/latest/translating/internationalization.html#backend-translations).

Thanks for the guidance. This fixed the test error. I will read up on Jinja!

timabbott commented:
@TanishaShrotriya96 please spent some time reading about ARIA -- you have `aria-hidden` on the same element as `aria-labelled-by` where I think you mean `aria-label`, which doesn't make sense.

See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA

I'm going to close this PR since it's pretty cluttered and I think you'd be better of starting over after learning more.
